assets/src/scripts/components/hydrograph/index.js

```diff
@@ -238,6 +233,126 @@ class Hydrograph {
             });
     }
 
+    _plotLegend() {
```

Comment:
Is all this more complicated than it needs to be?

Reply:
Seems like a reasonable approach to me. I didn't do a full review because it seems like you are still working on it.